### ISSUE TITLE
JsonSubTypes support always assumes a @type property

### ISSUE THREAD

**Issue by Legioth** [vaadin/hilla#1722]
### Describe the bug

The generator for `@JsonSubTypes` assumes that there will always be a property named `@type`. This is the case with the `@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY)` definition used with various built-in classes but it's not necessarily the case with other representations.

The property name can be customized with an optional `property` attribute on the `@JsonTypeInfo` annotation. If it's not defined, then the default value is based on the `use` value. `NAME` defaults to `@type` whereas e.g. `MINIMAL_CLASS` defaults to `@c` and `DEDUCTION` doesn't use any additional property at all.

### Expected-behavior

An identifying property is injected into the generated TypeScript only when Jackson would actually include one in the generated JSON values. When a property is injected, then the correct name is used for that property.

### Reproduction

Example without any additional property.
```
@JsonTypeInfo(use = JsonTypeInfo.Id.DEDUCTION)
@JsonSubTypes({ @Type(A.class), @Type(B.class) })
public class Base { }

public class A {
  private final String a;
  public A(String a) { this.a = a; }
  public String getA() { return a; }
}

public class B {
  private final String b;
  public B(String b) { this.b = b; }
  public String getB() { return b; }
}
```

An example with a different default property name can be seen by changing the built-in `Filter` to like this:
```
@JsonTypeInfo(use = JsonTypeInfo.Id.MINIMAL_CLASS, include = JsonTypeInfo.As.PROPERTY)
@JsonSubTypes({ @Type(value = OrFilter.class),
        @Type(value = AndFilter.class),
        @Type(value = PropertyStringFilter.class) })
public class Filter {

}
```

An example with a non-default property name can be seen by changing the `@JsonTypeInfo` of `Filter` like this:
```
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "t")
```

An example with a non-default type structure can be seen by changing the `@JsonTypeInfo` of `Filter` like this:
```
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.WRAPPER_OBJECT)
```

### System Info

Hilla 2.3.2

**Comment by cromoteca** [comment]
Supporting all the `JsonTypeInfo` options is a huge amount of work. I mean huge. That's why we chose to limit to one syntax.

At the same time, it should probably skip generation if the syntax is different.

**Comment by Legioth** [comment]
I agree. It's better to avoid injecting any property in the cases that are not explicitly supported since it's more convenient for the developer to include an additional property than to exclude a property that TypeScript thinks must be there.
